senet/se_googlenet

```diff
@@ -0,0 +1,49 @@
+from torch import nn
+from torchvision.models import googlenet
+from torchsummary import summary
+import warnings
+from collections import namedtuple
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.jit.annotations import Optional, Tuple
+from torch import Tensor
+from senet.se_module import SELayer
+
+class SEGoogLeNet(nn.Module):
+  def __init__(self, num_classes, pretrained, aux_logits=True, transform_input=False):
+    super(SEGoogLeNet, self).__init__()
+    #Pretrain and customize classifier
+    model = googlenet(pretrained = pretrained, aux_logits = aux_logits, transform_input = transform_input)
+    in_c = model.fc.in_features
+    model.fc = nn.Linear(in_c, num_classes)
+    # if aux_logits:
```

Comment:
Is `aux_logits` unused?

Reply:
I wasn't sure how to include the code for auxiliary heads since I had used the wrong `in_channel` in lines 23 and 24 where I was customizing it for a user-defined number of channels. I've corrected that now.

 I also had a divide by zero issue during kaiming initialization at the SE layers in the aux heads, which is caused when the channel of the SE layer is less than the reduction ratio. I've changed the ratio to be equal to the channel if channel < ratio. I'm not sure if this is the correct solution, but the model is now able to train.

By the way, how do I upload the changed file so you guys can see it? I'm new to open source, not really sure how to go about things yet.

Reply:
The original paper doesn't include SEGoogleNet, so if it can be trained, it can be SEGoogleNet. It's up to you.

Just push your changes to your branch `DLArchbold:patch-1`, then updates will be reflected here.

Reply:
I've cloned the fork DLArchbold/senet.pytorch so that I have a local repository, but I tried switching to the branch called "remotes/origin/patch-1", changed the se_googlenet file, committed, and tried pushing to origin/patch-1, but git says I'm not on a branch. I don't know how to proceed next.

Here's the image since github upload wasn't working.
https://imgur.com/a/DnTcY4q

Reply:
Hmm... How about `git push`?

Reply:
It still doesn't work and says I'm not currently on a branch. git push origin master also doesn't work.
https://imgur.com/a/LKAacs1

Reply:
Never mind, I think I figured out how it works and managed to push to the branch "patch-1".

Reply:
Yes, you made it!

Reply:
By the way, I would like to apologize for accidentally adding the branch called "se_googlenet" when trying to solve the problem.

Reply:
No worries!

Reply:
Hi, @moskomule , can this be called an open-source contribution to the official pytorch implementation of se_net?

Reply:
I’m not a member of the authors of the paper.

Reply:
Hmm, but the authors linked the PyTorch implementation to here.

Reply:
Hmm, but the authors linked the PyTorch implementation to here.